include/sys/statfs.h

```diff
@@ -114,6 +114,10 @@ struct statfs
   fsblkcnt_t f_bavail;   /* Free blocks avail to non-superuser */
   fsfilcnt_t f_files;    /* Total file nodes in the file system */
   fsfilcnt_t f_ffree;    /* Free file nodes in the file system */
+  struct
+  {
+    int val[2];
+  } f_fsid;              /* Encode device type, not yet in use */
```

Comment:
I think it is not common to have a struct defined inside other struct on NuttX, shouldn't it better to define f_fsid as f_fsid_s outside of statfs struct?

Reply:
>  I think it is not common to have a struct defined inside other struct on NuttX, shouldn't it better to define f_fsid as f_fsid_s outside of statfs struct?

Structures defined within other structures is actually very common. There are hundreds of examples. I would consider this a standard programming practice.

>  @acassis I defined the structure in types.h without typedef. is that okay?

No. Please do not modify sys/types.h. The content of that file is controlled by POSIX standards and should contain as few non-standard types as possible!!!

Reply:
> No. Please do not modify sys/types.h. The content of that file is controlled by POSIX standards and should contain as few non-standard types as possible!!!

@patacongo wheather I need to remove this structure from the sys/types.h file? if so, which would be the more appropriate file, I don't know much about it, thanks very much.